app/models/concerns/filterable.rb

```diff
@@ -0,0 +1,13 @@
+module Filterable
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def filter(filtering_params)
+      results = self.where(nil)
```

Comment:
so if i understand this method correctly, the way it works is results is initialized as all the records in the db for the model class, then within the loop a symbol representing a class method and a value representing an argument to that method are sent to the class the module is included on, causing the invocation of that method? why not use scopes or convert this method to a scope?